File: backend/app/api/exam_templates.py
"""資格試験テンプレートのAPI（実装フェーズ分割計画書Phase38）。"""

from fastapi import APIRouter

from app.schemas.exam_template import ExamTemplateRead
from app.services import exam_template_service

router = APIRouter(tags=["exam-templates"])


@router.get("/exam-templates", response_model=list[ExamTemplateRead])
def get_exam_templates() -> list[ExamTemplateRead]:
    return exam_template_service.list_exam_templates()

File: frontend/src/api/examTemplates.ts
import { apiClient } from './client'
import type { components } from '../types/api.d.ts'

export type ExamTemplateSubject = components['schemas']['ExamTemplateSubject']
export type ExamTemplateMaterial = components['schemas']['ExamTemplateMaterial']
export type ExamTemplateRead = components['schemas']['ExamTemplateRead']

/** 資格試験テンプレート一覧（仕様書「資格モードのテンプレート」）。 */
export function listExamTemplates(): Promise<ExamTemplateRead[]> {
  return apiClient.get<ExamTemplateRead[]>('/exam-templates')
}
